src/main/resources/react/src/components/pages/FormRegister/Form.js

```diff
@@ -0,0 +1,199 @@
+import React, { useEffect, useState } from "react";
+import { Redirect } from "react-router-dom";
+import Input from "./inputForForm/InputForForm.js";
+import { dataLogin, dataRegister, errorsServer } from "./inputBase";
+import { RegisterValidation } from "./validationInput";
+import RenderError from "./renderError/RenderError";
+import Button from "../../button/Button.js";
+import Extra from "./extra/Extra.js";
+import useFetch from "../hooks/doFetch";
+import useLocalStorage from "../hooks/useLocalStorage";
+
+import "./form.scss";
+
+const Form = props => {
+  const isLogin = props.match.path === "/registration/";
+  let arrayValueForm = isLogin ? dataLogin : dataRegister;
+  const oneClass = isLogin ? "login" : "register";
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [readyToEnter, setReadyToEnter] = useState(false);
+
+  const [isChecking, setIsChecking] = useState(false);
+
+  const [loginInputs, setLoginInputs] = useState({});
+  const [registerInputs, setRegisterInputs] = useState({});
+
+  const [errorsInput, setErrorsInput] = useState({});
+
+  const [{ isLoading, response, errorsResponse }, doFetch] = useFetch(
+    "auth",
+    oneClass
+  );
+  const [user, setUser] = useLocalStorage("user", { isLogin, isChecking });
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isSingIn, setIsSingIn] = useState(false);
+
+  const dataInputsState = isLogin ? loginInputs : registerInputs;
+  const setDataInputsState = isLogin ? setLoginInputs : setRegisterInputs;
+
+  useEffect(() => {
+    setIsSuccess(false);
+    setReadyToEnter(false);
+    setIsChecking(false);
+    setDataInputsState({});
+    setErrorsInput({});
+  }, [isLogin, setDataInputsState]);
+
+  useEffect(() => {
+    if (0 !== Object.keys(errorsInput).length) {
+      setReadyToEnter(false);
+      return;
+    }
+    const check = false;
+    if (
+      (0 === Object.keys(errorsInput).length &&
+        isChecking &&
+        checkAllInputs(check)) ||
+      isSubmitting
+    ) {
+      setReadyToEnter(true);
+    }
+    if (
+      (isLogin &&
+        0 === Object.keys(errorsInput).length &&
+        checkAllInputs(check)) ||
+      isSubmitting
+    ) {
+      setReadyToEnter(true);
+    }
+  }, [errorsInput, isChecking, isSubmitting]);
+
+  useEffect(() => {
+    if (!isSubmitting) {
+      return;
+    }
+    doFetch(dataInputsState);
+    setIsSubmitting(false);
+    setReadyToEnter(true);
+  }, [isSubmitting, dataInputsState]);
+
+  useEffect(() => {
+    if (!response) {
+      setIsSingIn(false);
+      return;
+    }
+    if (isLogin && isChecking) {
+      setUser(response);
+      setIsSingIn(true);
+    }
+    if (isLogin && !isChecking) {
+      setUser(response);
+      setIsSingIn(true);
```

Comment:
I observe here twice the same code that and ``isChecking`` do nothing useful here. It might be simplified to:
```
if (isLogin) {
      setUser(response);
      setIsSingIn(true);
    }
```

Reply:
did as you said, thanks